src/framework/core/js/FluidViewDebugging.js

```diff
@@ -0,0 +1,649 @@
+/*
+Copyright 2015 Lucendo Development Ltd.
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/* global console */
+
+var fluid_2_0 = fluid_2_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+    
+    fluid.registerNamespace("fluid.debug");
+    
+    fluid.debug.toggleClass = function (styles, element, openStyle, closedStyle, state) {
+        if (openStyle) {
+            element.toggleClass(styles[openStyle], state);
+        }
+        if (closedStyle) {
+            element.toggleClass(styles[closedStyle], !state);
+        }
+    };
+    
+    fluid.debug.bindToggleClick = function (element, applier, path) {
+        element.click(function () {
+            var state = fluid.get(applier.holder.model, path);
+            applier.change(path, !state);
+        });
+    };
+   
+    
+    fluid.defaults("fluid.debug.highlighter", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            highlightRoot: "#fluid-debug-highlightRoot"
```

Comment:
This container must be unique within the entire document - since it is used to house all highlight nodes even which are destined to hover over elements within nested iframes.